realbook/layers/signal.py

```diff
@@ -380,3 +380,151 @@ def get_config(self) -> Dict[str, Any]:
             }
         )
         return config
+    
+
+class Magnitude(tf.keras.layers.Layer):
+    def __init__(
+            self,
+            name: Optional[str] = None,
+            dtype: tf.dtypes.DType = tf.float32,
+            *args: Any,
+            **kwargs: Any,
+    ):
+        """
+        A Tensorflow Keras layer that calculates the magnitude of a complex tensor.
+        The input is complex-valued with shape (num_batches, time, fft_length // 2 + 1, 2).
+        The output is real-valued with shape (num_batches, time, fft_length // 2 + 1)
+
+        Args:
+            name: Name of the layer.
+            dtype: Type used in calculation.
+        """
+        super().__init__(name=name, dtype=dtype, *args, **kwargs)
+
+    def call(self, inputs: tf.Tensor) -> tf.Tensor:
+        return tf.math.abs(inputs)
+
+    def get_config(self) -> Dict[str, Any]:
+        config: Dict[str, Any] = super().get_config().copy()
+        return config
+
+
+class Phase(tf.keras.layers.Layer):
+    def __init__(
+            self,
+            name: Optional[str] = None,
+            dtype: tf.dtypes.DType = tf.float32,
+            *args: Any,
+            **kwargs: Any,
+    ):
+        """
+        A Tensorflow Keras layer that calculates the phase of a complex tensor.
+        The input is complex-valued with shape (num_batches, time, fft_length // 2 + 1, 2).
+        The output is real-valued with shape (num_batches, time, fft_length // 2 + 1)
+
+        Args:
+            name: Name of the layer.
+            dtype: Type used in calculation.
+        """
+        super().__init__(name=name, dtype=dtype, *args, **kwargs)
+
+    def call(self, inputs: tf.Tensor) -> tf.Tensor:
+        return tf.math.angle(inputs)
+
+    def get_config(self) -> Dict[str, Any]:
+        config: Dict[str, Any] = super().get_config().copy()
+        return config
+
+
+class MagnitudeToDecibel(tf.keras.layers.Layer):
+    def __init__(
+            self,
+            ref: float = 1.0,
+            amin: float = 1e-10,
+            top_db: Optional[float] = None,
+            name: Optional[str] = None,
+            dtype: tf.dtypes.DType = tf.float32,
+    ):
+        """
+        A Tensorflow Keras layer that converts magnitude spectrograms to decibel scale.
+        The input is real-valued with shape (num_batches, num_samples).
+        The output is real-valued with shape (num_batches, num_samples).
```

Comment:
```suggestion
        A Keras layer that converts a real-valued tensor to decibel scale.
```
This layer in theory doesn't require specifics about the shape. Can you rewrite it to remove any functionality related to shape?